src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -448,7 +448,7 @@ private async Task ExecuteInternal()
                 }
 
                 // Locate and setup the connection
-                queryConnection = await ConnectionService.Instance.GetOrOpenConnection(editorConnection.OwnerUri, ConnectionType.Query);
```

Comment:
It would be great to go and try to make sense of all this connection type stuff. Why do we even have it if we just ignore/override it on the STS side in some circumstances? And what do each of the values mean? It'll be too easy for any changes to this stuff to end up breaking some other completely unrelated thing if we aren't having a strict set of rules around this stuff...

(and then please document this in the wiki or somewhere so we actually have something to refer back to in the future)

Reply:
As of now, it seems the connections are always momentarily created for most scenarios (this is why I was interested in introducing connection pooling), however I'd need to thoroughly test all features to ensure that statement is true. I have briefly looked into primary touch areas as of now.

If connections are not falling in the 'closeable' category, features requesting connection need to own the lifecycle of the connection, as QE intends to do now.